src/charm.py

```diff
@@ -340,8 +348,12 @@ def _create_client(self, client_config, metadata=None):
 
         logger.info(cmd)
 
-        process = self._container.exec(cmd)
-        stdout, _ = process.wait_output()
+        try:
+            process = self._container.exec(cmd)
+            stdout, _ = process.wait_output()
+        except ExecError as err:
```

Comment:
I'm not 100% sure but I think the `ExecError` would only be raised on the `wait_output` call so I'd remove the `exec` from the try/except block

Reply:
You are right, `exec` can only throw errors when the operator fails to connect to the pebble websocket. Is it ok now or do should we split the try/except?

Reply:
I'd put only the `wait_output` inside of the try/except block and place the `exec` right before